lua/obsidian/collections.lua

```diff
@@ -91,4 +91,39 @@ end
 --   return self
 -- end
 
+-------------------------
+--- Stack implementation
+-------------------------
+
+-- Create new stack
+function M.new_stack()
+  return {}
+end
+
+-- Push an item onto the stack
+function M.push(stack, item)
+  table.insert(stack, item)
+end
+
+-- Pop an item from the stack
+function M.pop(stack)
+  if #stack == 0 then
+    return nil
+  end
+  return table.remove(stack)
+end
+
+-- Check the top item on the stack
+function M.peek(stack)
+  if #stack == 0 then
+    return nil
+  end
+  return stack[#stack]
+end
+
+-- Check if the stack is empty
+function M.is_empty(stack)
+  return #stack == 0
+end
```

Comment:
This isn't really how Lua classes are implemented, and it doesn't seem necessary to have a separate class here anyway since it's just a very thin wrapper around a raw table/list.

Reply:
I'm fine with removing this all together then.